src/TooltipPopoverWrapper.js

```diff
@@ -231,54 +250,55 @@ class TooltipPopoverWrapper extends React.Component {
           document.addEventListener('click', this.handleDocumentClick, true);
         }
 
-        if (this._target) {
+        if (this._targets && this._targets.length) {
           if (triggers.indexOf('hover') > -1) {
-            this._target.addEventListener(
+            this.addEventOnTargets(
               'mouseover',
               this.showWithDelay,
               true
             );
-            this._target.addEventListener(
+            this.addEventOnTargets(
               'mouseout',
               this.hideWithDelay,
               true
             );
           }
           if (triggers.indexOf('focus') > -1) {
-            this._target.addEventListener('focusin', this.show, true);
-            this._target.addEventListener('focusout', this.hide, true);
+            this.addEventOnTargets('focusin', this.show, true);
+            this.addEventOnTargets('focusout', this.hide, true);
           }
-          this._target.addEventListener('keydown', this.onEscKeyDown, true);
+          this.addEventOnTargets('keydown', this.onEscKeyDown, true);
         }
       }
     }
   }
 
   removeTargetEvents() {
-    if (this._target) {
-      this._target.removeEventListener(
+    if (this._targets) {
+      this.removeEventOnTargets(
         'mouseover',
         this.showWithDelay,
         true
       );
-      this._target.removeEventListener(
+      this.removeEventOnTargets(
         'mouseout',
         this.hideWithDelay,
         true
       );
-      this._target.removeEventListener('keydown', this.onEscKeyDown, true);
-      this._target.removeEventListener('focusin', this.show, true);
-      this._target.removeEventListener('focusout', this.hide, true);
+      this.removeEventOnTargets('keydown', this.onEscKeyDown, true);
+      this.removeEventOnTargets('focusin', this.show, true);
+      this.removeEventOnTargets('focusout', this.hide, true);
     }
 
     document.removeEventListener('click', this.handleDocumentClick, true)
   }
 
   updateTarget() {
-    const newTarget = getTarget(this.props.target);
-    if (newTarget !== this._target) {
+    const newTarget = getTarget(this.props.target, true);
```

Comment:
This PR broke support for refs as `target` prop. When `props.target` is a React ref or an HTMLElement, this will return an HTMLElement, not an array. `Array.from(newTarget)` will then return an empty array, and Popper will error that `target` was set to `undefined`.

Reply:
fixed in #1687